field.go

```diff
@@ -187,14 +187,18 @@ func Error(err error) zapcore.Field {
 // extremely expensive (relatively speaking); this function both makes an
 // allocation and takes ~10 microseconds.
 func Stack(key string) zapcore.Field {
+	return stack(key, 4)
```

Comment:
That is exported function and called outside of package too. 
So `skip` should be 1 by default, but `Logger` should use its `callerSkip` to define caller stack frame. 
Also consider add  exported`StackSkip(key string, skip int) zapcore.Field`, to allow caller skip some frames. It's useful in  recover: skip allows set top stack frame panic caller.